### PR TITLE
[Call Automation] 2024-04-15 GA3 Release

### LINKED ISSUE #27878: [IC3 ACS Server Calling ] API Review
New API Review meeting has been requested.

**Service Name**: IC3 ACS Server Calling 
**Review Created By**: Lei Xu
**Review Date**: 03/07/2024 08:00 AM PT
**Release Plan**: [](https://web.powerapps.com/apps/3a732482-e479-4ec1-99c8-359f2c2d3f53?release-plan-id=)
**PR**: 
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=871db7a1-b7d1-ee11-9079-000d3a3412d3)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by lei-msft**
# Data Plane API - Pull Request
Azure Communication Services Call Automation provides developers the ability to build server-based, intelligent call workflows, and call recording for voice and PSTN channels.

This is for the third GA release for ACS's Call Automation.

## API Info: The Basics
Start Recording: Allowed to start a recording in Pause mode before customer joins. In addition, users are given an option to use their own blob storage for storing recordings

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [X] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>

The GA3 includes three features regarding Recording

1. PauseOnStart
2. TeamsComplianceRecording
3. BYOS (Bring Your Own Storage)

* Design Document:
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/CallAutomation/stable/2023-10-15/communicationservicescallautomation.json
* Updated paths: StartRecording

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

File: specification/communication/data-plane/CallAutomation/stable/2024-04-15/examples/CallRecording_StartRecording.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        "kind": "serverCallLocator"
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "recordingId": "eyJQbGF0Zm9ybUVuZHBvaW50SWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUiLCJSZXNvdXJjZVNwZWNpZmljSWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUifQ==",
+        "recordingState": "inactive",
+        "recordingKind": "teamsCompliance"
```
Comment (khannavikas) [inline specification/communication/data-plane/CallAutomation/stable/2024-04-15/examples/CallRecording_StartRecording.json:28]: Start recording will always have recording kind as "AzureCommunicationService"

File: specification/communication/data-plane/CallAutomation/stable/2024-04-15/examples/CallRecording_GetRecordingProperties.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "parameters": {
+    "endpoint": "https://contoso.communications.azure.com",
+    "api-version": "2024-04-15",
+    "recordingId": "eyJQbGF0Zm9ybUVuZHBvaW50SWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUiLCJSZXNvdXJjZVNwZWNpZmljSWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUifQ=="
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "recordingId": "eyJQbGF0Zm9ybUVuZHBvaW50SWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUiLCJSZXNvdXJjZVNwZWNpZmljSWQiOiI4YjlkYmFkMS0xOGFiLTRmODYtYWQ0Yy1jNzVmODA5ODA3NmUifQ==",
+        "recordingState": "inactive"
```
Comment (khannavikas) [inline specification/communication/data-plane/CallAutomation/stable/2024-04-15/examples/CallRecording_GetRecordingProperties.json:11]: We need to have recordingKind here

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/communication/data-plane/CallAutomation/stable/2024-04-15/communicationservicescallautomation.json [hunk 3]
```diff
@@ -0,0 +1,3076 @@
...
+        "teams",
+        "teamsCompliance",
+        "others"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingKind",
+        "modelAsString": true
+      }
+    },
```
Comment (mikekistler) [inline specification/communication/data-plane/CallAutomation/stable/2024-04-15/communicationservicescallautomation.json:2315]: You can include "values" in the "x-ms-enum" with a description for each value, and these descriptions will then be shown in the generated REST API Docs. Descriptions for enum values will improve the developer experience for your service. See the [autorest documentation][] for the syntax of the "values" field in "x-ms-enum".

[autorest documentation]: https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-enum

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**